lib/app/features/user/pages/profile_page/components/tabs/tab_entities_list.dart

```diff
@@ -3,25 +3,50 @@
 import 'package:flutter/material.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:ion/app/components/scroll_view/load_more_builder.dart';
+import 'package:ion/app/features/auth/providers/auth_provider.c.dart';
 import 'package:ion/app/features/components/entities_list/entities_list.dart';
 import 'package:ion/app/features/components/entities_list/entities_list_skeleton.dart';
 import 'package:ion/app/features/nostr/model/nostr_entity.dart';
 import 'package:ion/app/features/nostr/providers/entities_paged_data_provider.c.dart';
+import 'package:ion/app/features/user/model/tab_entity_type.dart';
 import 'package:ion/app/features/user/pages/profile_page/components/tabs/empty_state.dart';
+import 'package:ion/app/features/user/providers/tab_data_source_provider.c.dart';
+import 'package:ion/app/features/user/providers/user_metadata_provider.c.dart';
+import 'package:ion/app/utils/username.dart';
 
 class TabEntitiesList extends ConsumerWidget {
-  const TabEntitiesList({
-    required this.dataSource,
+  factory TabEntitiesList({
+    required String pubkey,
+    required TabEntityType type,
+    Key? key,
+  }) {
+    return TabEntitiesList._(
+      type: type,
+      pubkey: pubkey,
+      builder: type == TabEntityType.replies
+          ? (entities) => EntitiesList(entities: entities.toList(), showParent: true)
+          : null,
+      key: key,
+    );
+  }
```

Comment:
Maybe better create separate factory constructor for replies? ` factory TabEntitiesList.replies`

Reply:
Done, good idea.